arelle/plugin/validate/DBA/rules/fr.py

```diff
@@ -71,6 +72,47 @@ def rule_fr7(
     )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_fr34(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.FR34: If Equity does not equal 0 more fields are required.  At least one field must be filled in on the balance sheet in addition to equity:
+    Assets, NoncurrentAssets, CurrentAssets, LongtermLiabilitiesOtherThanProvisions, ShorttermLiabilitiesOtherThanProvisions, LiabilitiesOtherThanProvisions, LiabilitiesAndEquity
+    """
+    equityFacts = val.modelXbrl.factsByQname.get(pluginData.equityQn)
+    otherRequiredFactsQnames = []
+    nonZeroEquityFacts = []
+    otherRequiredFacts = []
+    if equityFacts:
+        for fact in equityFacts:
+            if fact is not None and fact.xValid <=VALID:
```

Comment:
Why is the `fact is not None` necessary?

Why `<= VALID`?  Normally we use `>= VALID` to ensure `xValue` is non-`None` (I guess in this case it's fine either way since `None != 0` is fine, but it seems clearer to guard all uses of `xValue`?).